File: 2043. Simple Bank System.py
class Bank:

    def __init__(self, balance: List[int]):
        self.balance = {}

        for i, bal in enumerate(balance):
           self.balance[i + 1] = bal 

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if 1 <= account1 and account1 <= len(self.balance) and 1 <= account2 and account2 <= len(self.balance) and self.balance[account1] >= money:
            self.balance[account1] -= money
            self.balance[account2] += money
            return True

        return False

    def deposit(self, account: int, money: int) -> bool:
        if 1 <= account and account <= len(self.balance):
            self.balance[account] += money
            return True

        return False

    def withdraw(self, account: int, money: int) -> bool:
        if 1 <= account and account <= len(self.balance) and self.balance[account] >= money:
            self.balance[account] -= money
            return True

        return False


# Your Bank object will be instantiated and called as such:
# obj = Bank(balance)
# param_1 = obj.transfer(account1,account2,money)
# param_2 = obj.deposit(account,money)
# param_3 = obj.withdraw(account,money)

File: 2043. Simple Bank System.cpp
class Bank {
public:
    Bank(vector<long long>& balance) : balance(balance){
        
    }
    
    bool transfer(int account1, int account2, long long money) {
        if((1 <= account1 && account1 <= balance.size()) && (1 <= account2 && account2 <= balance.size()) && balance[account1 - 1] >= money){
            balance[account1 - 1] -= money;
            balance[account2 - 1] += money;
            return true;
        }

        return false;
    }
    
    bool deposit(int account, long long money) {
        if((1 <= account && account <= balance.size())){
            balance[account - 1] += money;
            return true;
        }

        return false;
    }
    
    bool withdraw(int account, long long money) {
        if((1 <= account && account <= balance.size()) && balance[account - 1] >= money){
            balance[account - 1] -= money;
            return true;
        }

        return false;
    }
private:
    vector<long long>& balance;
};

/**
 * Your Bank object will be instantiated and called as such:
 * Bank* obj = new Bank(balance);
 * bool param_1 = obj->transfer(account1,account2,money);
 * bool param_2 = obj->deposit(account,money);
 * bool param_3 = obj->withdraw(account,money);
 */
